src/service/scenarios/delete_scenarios_service.py

```diff
@@ -0,0 +1,237 @@
+
+    def delete_scenario_from_details(self, scenario_id: str, metric_id) -> bool:
+        metric_is_deleted = self.delete_metric(scenario_id, metric_id)
+        if metric_is_deleted:
+            is_scenario = self.scenario_is_in_scenario_metric(scenario_id)
+            if not is_scenario:
+                period_id = self.get_time_period_of_scenario(scenario_id)
+                scenario_deleted = False
+                try:
+                    query = """
+                    DELETE FROM {table_name} WHERE id = '{scenario_id}'
+                    """.format(
+                        table_name="financial_scenario", scenario_id=scenario_id
+                    )
+
+                    self.session.execute(query)
+                    self.session.commit()
+                    scenario_deleted = True
+                except Exception as error:
+                    self.logger.info(error)
+                    scenario_deleted = False
+
+                if scenario_deleted:
+                    self.delete_time_period(period_id)
+                    return True
+            else:
+                return True
+        else:
+            return False
+
+    def delete_scenario(
+        self, name, year, company_id, metric_name, metric_value
+    ) -> bool:
+        print(name, year, company_id, metric_name, metric_value)
+        metric_id = self.get_metric_id(metric_name, company_id, metric_value)
+        scenario_id = self.get_scenario_id(name, year, company_id)
+        metric_is_deleted = self.delete_metric(scenario_id, metric_id)
+        print(metric_id)
+        print(scenario_id)
+        if metric_is_deleted:
+            is_scenario = self.scenario_is_in_scenario_metric(scenario_id)
+            if not is_scenario:
+                period_id = self.get_time_period_of_scenario(scenario_id)
+                scenario_deleted = False
+                try:
+                    query = """
+                    DELETE FROM {table_name} WHERE id = '{scenario_id}'
+                    """.format(
+                        table_name="financial_scenario", scenario_id=scenario_id
+                    )
+
+                    self.session.execute(query)
+                    self.session.commit()
+                    scenario_deleted = True
+                except Exception as error:
+                    self.logger.info(error)
+                    scenario_deleted = False
+
+                if scenario_deleted:
+                    self.delete_time_period(period_id)
+                    return True
+            else:
+                return True
+        else:
+            return False
+
+    def get_time_period_of_scenario(self, scenario_id: str):
+        try:
+            query = """
+            SELECT period_id FROM {table_name} WHERE id = '{scenario_id}'
+            """.format(
+                table_name="financial_scenario", scenario_id=scenario_id
+            )
+
+            result = self.session.execute(query)
```

Comment:
We're using fetchall() instead of first()
And then use response_sql to parse the result if it is needed, like in the other services that we have.